src/target/km1m7xx.c

```diff
@@ -0,0 +1,1965 @@
+
+	if (CMD_ARGC != 1)
+		return ERROR_COMMAND_SYNTAX_ERROR;
+
+	if (!target) {
+		LOG_ERROR("no target selected");
+		return ERROR_FAIL;
+	}
+
+	image.base_address_set	= 0;
+	image.base_address		= 0;
+	image.start_address_set	= 0;
+
+	retval = image_open(&image, CMD_ARGV[0], NULL);
+	if (retval != ERROR_OK)
+		return retval;
+
+	image_size = 0x0;
+	retval = ERROR_OK;
+	for (i = 0; i < image.num_sections; i++) {
+		buffer = malloc(image.sections[i].size);
+		if (buffer == NULL) {
+			command_print(CMD,
+					"error allocating buffer for section (%d bytes)",
+					(int)(image.sections[i].size));
+			break;
+		}
+		retval = image_read_section(&image, i, 0x0, image.sections[i].size, buffer, &buf_cnt);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		/* calculate checksum of image */
+		retval = image_calculate_checksum(buffer, buf_cnt, &checksum);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		retval = target_checksum_memory(target, image.sections[i].base_address, buf_cnt, &mem_checksum);
+		if (retval != ERROR_OK) {
+			free(buffer);
+			break;
+		}
+
+		free(buffer);
+		image_size += buf_cnt;
+	}
+
+	return retval;
+}
+
+COMMAND_HANDLER(km1m7xx_handle_keycode_file_command)
+{
+	FILE		*fp_keyfile;
+	char		key_str[16];
+	int			key_count;
+
+	if (CMD_ARGC != 1) {
+		return(ERROR_COMMAND_SYNTAX_ERROR);
+	}
+
+	fp_keyfile = fopen(CMD_ARGV[0], "r");
+	if (fp_keyfile == NULL) {
+		return(ERROR_FAIL);
+	}
+
+	key_count = 0;
+	while (fgets(key_str, 15, fp_keyfile) != NULL) {
+		km1m7xx_key_data[key_count++] = strtoul(key_str, NULL, 16);
+	}
+	fclose(fp_keyfile);
+
+	km1m7xx_key_set = 1;
+	
+	return(ERROR_OK);
+}
+
+COMMAND_HANDLER(km1m7xx_handle_keycode_data_command)
+{
+	char		key_str[16];
+	int			key_count;
+
+	if (CMD_ARGC != 4) {
+		return(ERROR_COMMAND_SYNTAX_ERROR);
+	}
+
+	for (key_count = 0; key_count < 4; key_count++) {
+		if (strncmp(CMD_ARGV[key_count], "0x", 2) != 0) {
+			strcpy(key_str, "0x");
+		} else {
+			key_str[0] = '\0';
+		}
+		strcat(key_str, CMD_ARGV[key_count]);
+		COMMAND_PARSE_NUMBER(u32, key_str, km1m7xx_key_data[key_count]);
+	}
+
+	km1m7xx_key_set = 1;
+	return(ERROR_OK);
+}
+
+static const struct command_registration km1m7xx_subcommand_handlers[] = {
+	{
+		.name		= "calc_image_checksum",
+		.handler	= km1m7xx_handle_calc_image_checksum_command,
+		.mode		= COMMAND_ANY,
+		.usage		= "<file>",
+		.help		= "calculate checksum of image file",
+	},
+	{
+		.name		= "calc_memory_checksum",
+		.handler	= km1m7xx_handle_calc_memory_checksum_command,
+		.mode		= COMMAND_EXEC,
+		.usage		= "<file>",
+		.help		= "calculate checksum of target memory and image file",
+	},
+	{
+		.name		= "keycode_file",
+		.handler	= km1m7xx_handle_keycode_file_command,
+		.mode		= COMMAND_CONFIG,
+		.usage		= "<file>",
+		.help		= "Set keycode file for authentication",
+	},
+	{
+		.name		= "keycode_data",
+		.handler	= km1m7xx_handle_keycode_data_command,
+		.mode		= COMMAND_CONFIG,
+		.usage		= "<file>",
```

Comment:
Please give a meaningful usage message if the parameter count is larger than one.

Reply:
Corrected the writing style of ".usage=" like other files. (Listed parameters)